osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderRepeat.cs

```diff
@@ -85,8 +87,18 @@ protected override void OnApply()
 
         protected override void CheckForResult(bool userTriggered, double timeOffset)
         {
-            if (HitObject.StartTime <= Time.Current)
-                ApplyResult(r => r.Type = DrawableSlider.Tracking.Value ? r.Judgement.MaxResult : r.Judgement.MinResult);
+            // shared implementation with DrawableSliderTick.
+            if (timeOffset >= 0)
+            {
+                // Attempt to preserve correct ordering of judgements as best we can by forcing
+                // an un-judged head to be missed when the user has clearly skipped it.
+                //
+                // This check is applied to all nested slider objects apart from the head (ticks, repeats, tail).
+                if (Tracking && !DrawableSlider.HeadCircle.Judged)
```

Comment:
Why only if tracking? If not tracking, then wouldn't the repeat get judged as a min result first?

Reply:
It will, but in general a min result won't lead to a difference in scoring from mis-ordering.

Why don't we change this to always trigger the miss though? That's because if we did, it would make it impossible to hit the slider head late on a very fast repeat slider (or when a tick occurs very soon after the head) reducing the hit window and feeling quite bad to the user.

The main this we're trying to avoid here is having a tick *hit* occur without the head miss preceding it.

Reply:
So you're saying that if you miss a slider repeat, you can then start tracking the slider and hit the slider head, and that makes sense to you?

Reply:
Timing wise that would be unfeasible.

Think of this from a player's perspective, either:

- They are going to hit the head and hit it late
- They missed the head and have started tracking, hitting the first tick/repeat (and with this change, triggering the miss of the head early to keep more correct ordering)